### ISSUE TITLE
Spacemacs layouts - SPC b b not layout local if invoked before SPC B b if ido is excluded

### ISSUE THREAD

**Issue by cpaulik** [syl20bnr/spacemacs#5788]
#### Description
`SPC b b` is not layout local until `SPC B b` was invoked once when `ido` is in `dotspacemacs-excluded-packages`

#### Reproduction guide
- Start Emacs
- `SPC p l` `SPC b b`. This shows all buffers for me. 
- Then `SPC B b` `ESC` `SPC b b` shows only the buffers of the project.

*Expected behaviour:*
The first `SPC b b` should only show the layout buffer.

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.18
- Spacemacs branch: develop (rev. 5ee970f)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(ansible
 (auto-completion :variables auto-completion-enable-snippets-in-popup t auto-completion-private-snippets-directory "~/dotfiles/.spacemacs-config/snippets/" auto-completion-enable-help-tooltip nil)
 chrome clojure colors custom_org_config django
 (elfeed :variables rmh-elfeed-org-files
         (list "~/Dropbox/org/feeds.org"))
 emacs-lisp emoji erc finance games
 (git :variables git-gutter-use-fringe t)
 github gnus graphviz html ipython-notebook javascript latex lua markdown my-mu4e my_python org org-ipython org-page pandoc pdf-tools php
 (python :variables python-enable-yapf-format-on-save nil python-auto-set-local-pyenv-version 'on-project-switch python-test-runner
         '(pytest))
 ranger research-config restclient ruby rust search-engine semantic
 (shell :variables shell-default-shell 'term shell-default-term-shell "/bin/zsh")
 shell-scripts
 (spell-checking :variables spell-checking-enable-by-default nil)
 spacemacs-helm spacemacs-layouts syntax-checking systemd tmux vagrant
 (version-control :variables version-control-diff-tool 'diff-hl)
 yaml)
```

#### Backtrace
```
BACKTRACE IF RELEVANT
```

**Comment by StreakyCobra** [comment]
It seems to still be layout local here. But it's late, maybe I did something wrong. Will test further tomorrow.

**Comment by cpaulik** [comment]
Hmm, it is also late here. I've tried again now. It starts working correctly when I press `SPC B b` once. So please try from a freshly started emacs `SPC p l` `SPC b b`. This shows all buffers for me. Then `SPC B b` `ESC` `SPC b b` shows only the buffers of the project.

**Comment by robbyoconnor** [comment]
I've yet to be able to get it to work

**Comment by StreakyCobra** [comment]
I tested again, I still cannot reproduce. This is what I did:

### Preparation

- Rebased on last `develop`
- Updated packages
- `rm ~/.emacs.d/elpa/pers-mode…` just to be sure it's not quelpa version anymore
- Started Emacs to redownload persp-mode, closed emacs.

### Reproducing

- Starting Emacs
- `SPC p l`, selecting a project, pressing `RET`, selecting a file, pressing `RET`
- `SPC b b`

This shows me only the file I have selected in the current project, `*messages*` and `*spacemacs*` are not shown for instance. `SPC B b` gives the complete list.

**Comment by cpaulik** [comment]
@StreakyCobra Thanks. I've seen this on two machines of mine. I'll try to wipe the complete elpa when I have time to see if it persists.

**Comment by sooheon** [comment]
By the way, for those using `spacemacs-ivy` layer this functionality is broken, as `helm-mini` is hardcoded in the function.

**Comment by cpaulik** [comment]
I've completely wiped elpa and reinstalled everything. Still the same :worried:

**Comment by syl20bnr** [comment]
Mmhhhh did you try to remove m the Spacemacs-layouts from the list of your used layers ? This is a shot in the dark I'm not in front of a computer.

Did you try with a vanilla Spacemacs ?

**Comment by robbyoconnor** [comment]
Are you using helm or ivy?

**Comment by robbyoconnor** [comment]
`SPC l b` is correct -- and layout local

**Comment by robbyoconnor** [comment]
False alarm

**Comment by cpaulik** [comment]
Thanks, guys. It was caused by having `ido` in my `dotspacemacs-excluded-packages`. I don't remember why exactly I put it there. I guess it was used somewhere where I wanted helm.

**Comment by cpaulik** [comment]
Should I close this or does this require a fix?

**Comment by syl20bnr** [comment]
Keep it open for now, I don't understand why excluding ido would trigger this bug.

**Comment by deuill** [comment]
Not sure if I'm understanding this bug report wrong, but are buffers supposed to be layout-specific (as proposed in #4590)? If so, I'm getting the same issue, i.e.:

  * Start up Emacs
  * Change to project layout with `SPC p l`, choose any project file (say, `main.c`)
  * List of buffers with `SPC b b` includes `*spacemacs*`, `*Messages*` etc
  * Change to different project layout, choose any file. List of layouts (`SPC b b`) includes files from previous project layout (in this case, `main.c`).

Restarting Emacs also stores all buffers across layouts. My excluded packages are empty.

**Comment by StreakyCobra** [comment]
I'm not sure if the layout-specific feature is in `master` yet. On which branch are you?

**Comment by deuill** [comment]
@StreakyCobra ah, my bad, I am indeed on `master`. I missed the part where the bug has been reported against `develop`. Consider this my +1 though!

**Comment by StreakyCobra** [comment]
:smiley: It will arrive in `master` in the next release

**Comment by sooheon** [comment]
Is there a milestone for spacemacs-ivy compatibility, (and someone interested in working on it) or does it need a volunteer?

**Comment by d12frosted** [comment]
Fixed with release of Spacemacs v0.200.

**Comment by robbyoconnor** [comment]
Not so fast: #7240 does this but it's not fixed yet.
